src/enzyme_ad/jax/Implementations/HLODerivatives.td

```diff
@@ -536,6 +536,30 @@ def : HLOReadOnlyIdentityOp<"GatherOp", [0]>;
 
 def : HLOInactiveOp<"CeilOp">;
 
+def : HLODerivative<"ClampOp", (Op $min, $operand, $max), [
+  (Select
+    (And
+      (Compare $min, $max, (LT)),
+      (Compare $min, $operand, (GT))
+    ),
+    (DiffeRet),
+    (HLOConstantFP<"0"> $min)
+  ),
+  (Select
+    (Or
+      (Compare $operand, $min, (LT)),
+      (Compare $operand, $max, (GT))
+    ),
+    (HLOConstantFP<"0"> $operand),
+    (DiffeRet)
+  ),
+  (Select
+    (Compare (Max $operand, $min), $max, (GT)),
+    (DiffeRet),
+    (HLOConstantFP<"0"> $max)
+  ),
+]>;
```

Comment:
This feels like it should have a custom forward mode rule for efficiency rather than sum up the individual operands [you can add this as an extra arg after the list]